src/poly/dcrt/matrix/mmap_matrix.rs

```diff
@@ -235,30 +287,30 @@ impl<T: MmapMatrixElem> MmapMatrix<T> {
         for (idx, other) in others.iter().enumerate() {
             if self.nrow != other.nrow {
                 panic!(
-                    "Concat error: while the shape of the first matrix is ({}, {}), that of the {}-th matrix is ({},{})",
+                    "Concat error: while the shape of the first matrix is ({}, {}), \
+                     that of the {}-th matrix is ({},{})",
                     self.nrow, self.ncol, idx, other.nrow, other.ncol
                 );
             }
         }
         let updated_ncol = others.iter().fold(self.ncol, |acc, other| acc + other.ncol);
         let mut new_matrix = Self::new_empty(&self.params, self.nrow, updated_ncol);
-        let self_f = |row_offsets: Range<usize>, col_offsets: Range<usize>| -> Vec<Vec<T>> {
-            self.block_entries(row_offsets, col_offsets)
-        };
-        new_matrix.replace_entries(0..self.nrow, 0..self.ncol, self_f);
-        debug_mem("self replaced in concat_columns");
-
-        let mut col_acc = self.ncol;
-        for (idx, other) in others.iter().enumerate() {
-            let other_f = |row_offsets: Range<usize>, col_offsets: Range<usize>| -> Vec<Vec<T>> {
-                let col_offsets = col_offsets.start - col_acc..col_offsets.end - col_acc;
-                other.block_entries(row_offsets, col_offsets)
-            };
-            new_matrix.replace_entries(0..self.nrow, col_acc..col_acc + other.ncol, other_f);
-            debug_mem(format!("the {}-th other replaced in concat_columns", idx));
-            col_acc += other.ncol;
+        let self_columns = self.block_entries_column(0..self.nrow, 0..self.ncol);
+        let mut combined_columns = self_columns;
+        for other in others {
+            let other_columns = other.block_entries_column(0..self.nrow, 0..other.ncol);
+            combined_columns.extend(other_columns);
```

Comment:
I think this implementation might put too much amount of data to physical RAM because it does not ensure that the data loaded into RAM at the same time is bounded by `num_threads * block_size * block_size` elements.

Reply:
For example, how about combining the write operations for two (and more) matrices if the total number of elements are less than `block_size * block_size`?

Reply:
actually, I'm not sure bcs, currently, the boottleneck we are facing is:
1. `Too many open files` which we need to reduce usage of mmap 
2. the time is generally too slow -- good part is that memory is really bounded with mmap. 

So if this implementation is reducing the usage of I/O call + also speed up the time by using RAM, I think it is better approach for consider current situation

Reply:
i've confirm `Too many open files` resolve on this branch implemenation